Assets/Scripts/EOSSessionsManager.cs

```diff
@@ -331,6 +345,81 @@ public override int GetHashCode()
         {
             return base.GetHashCode();
         }
+
+        /// <summary>
+        /// Determines if the local user owns this Session.
+        /// This information can be used to gate some Session modifying actions.
+        /// </summary>
+        /// <returns>
+        /// True if the Session is owned by the local user.
+        /// False if either it was determined that the user doesn't own this Session,
+        /// or if the operation to query that information failed.
+        /// </returns>
+        public bool IsLocalUserOwnerOfSession
+        {
+            get
+            {
+                ProductUserId localUserId = EOSManager.Instance?.GetProductUserId();
+
+                // If we can't even get the local user's id, that state suggests they shouldn't be owning any Sessions
+                if (localUserId == null)
+                {
+                    return false;
+                }
+
+                // If the ActiveSession isn't set, then this user isn't a part of the Session
+                // Can't possibly own it
+                if (ActiveSession == null)
+                {
+                    return false;
+                }
+
+                ActiveSessionCopyInfoOptions options = new ActiveSessionCopyInfoOptions() { };
+                Result copyResult = ActiveSession.CopyInfo(ref options, out ActiveSessionInfo? sessionInfo);
+
+                if (copyResult != Result.Success || !sessionInfo.HasValue || !sessionInfo.Value.SessionDetails.HasValue)
+                {
+                    return false;
+                }
+
+                return sessionInfo.Value.SessionDetails.Value.OwnerUserId.Equals(localUserId);
```

Comment:
This is a lot of work to do within a Property field member. The expectation is that Properties should have a "runtime" as close to a bare field member as possible. I think you should either refactor this property to cache the value returned so that it is only retrieved once, or change it from a Property to a static function with the signature:
```csharp
public static bool DoesLocalUserOwnSession(ActiveSession session)
{
    /* ... */
}
```

You could also make the above `static` function private, and call it from a public instance function or something.

Reply:
I'm comfortable with the idea that this is too big for a property. It's not *expensive* work, but it is a lot of logic.

I didn't want to cache the value, in the case that this is accessed before the Session has had all of its information saturated properly... but upon looking at this, I realized that I don't have to actually store the cached value until the operation gets to a comfortable point. If ActiveSession etc is null, just return false without hitting the cache.

Reply:
This function is part of the Session object already, so I don't need to pass in the parameter.

Reply:
I also feel it is more appropriate for this to be a method or static method instead of a property.